### PR TITLE
Rename(LocalVariables|PrivateFields)ToCamelCase do not replace _ if next character is not an alphabetic char

### PR DESCRIPTION
**Pull request by pstreef**
fixes: https://github.com/openrewrite/rewrite-static-analysis/issues/103
Add test and proposed fix

### CODE REVIEW

File: src/main/java/org/openrewrite/staticanalysis/RenameLocalVariablesToCamelCase.java [hunk 1]
```diff
@@ -72,6 +72,12 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new RenameToCamelCase() {
             @Override
             protected boolean shouldRename(Set<String> hasNameKey, J.VariableDeclarations.NamedVariable variable, String toName) {
+                if(toName.isEmpty()) {
+                    return false;
+                }
+                if(!Character.isAlphabetic(toName.charAt(0))) {
+                    return false;
+                }
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/staticanalysis/RenameLocalVariablesToCamelCase.java:80]:
```suggestion
                if (toName.isEmpty() || !Character.isAlphabetic(toName.charAt(0))) {
                    return false;
                }
```

**Review by timtebeek — APPROVED** [review]
Thanks again! Let's merge as soon as the tests pass after merging https://github.com/openrewrite/rewrite/pull/3118

**Comment by pstreef** [comment]
I left the 2 checks split as maybe we want to warn in the first case, but forgot to add a todo. I'm not sure if that is also part of the "philosophy" of openrewrite?

**Comment by timtebeek** [comment]
> I left the 2 checks split as maybe we want to warn in the first case, but forgot to add a todo. I'm not sure if that is also part of the "philosophy" of openrewrite?

We generally don't add a lot of warnings, as recipes are often ran together in batches, and would otherwise flood the output with warnings. In this case it's fine to back off from a rename if the resulting name would be empty. We'd just not update the code in those cases, which is the safe thing to do.

Any final thoughts @knutwannheden ? You seem to have added (or extracted) `RenameToCamelCase` before.

File: src/main/java/org/openrewrite/staticanalysis/RenameLocalVariablesToCamelCase.java [hunk 2]
```diff
@@ -72,6 +72,9 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new RenameToCamelCase() {
             @Override
             protected boolean shouldRename(Set<String> hasNameKey, J.VariableDeclarations.NamedVariable variable, String toName) {
+                if (toName.isEmpty() || !Character.isAlphabetic(toName.charAt(0))) {
```
[the remark this answers is not in the document]
Comment (knutwannheden) [reply src/main/java/org/openrewrite/staticanalysis/RenameLocalVariablesToCamelCase.java:75]: Yes, your change should probably also do it. Let's do it like this first.

**Pull request merged**